types/lib/amazon-pay.d.ts

```diff
@@ -0,0 +1,46 @@
+import { Emitter } from './emitter';
+
+export type AmazonPayOptions = {
+  /**
+   * 2 Digit Country Code
+   */
+  region: string;
+
+  /**
+   * Specify which Payment Gateway in Recurly must handle the payment.
+   */
+  gatewayCode?: string
+
+  /**
+   * Sets button to Sandbox environment
+   */
+  sandbox?: boolean;
+};
+
+export type AmazonPayRenderButtonOptions = {
+  productType?: string;
+  placement?: string;
+  buttonColor?: string;
+};
+
+export type AmazonPayEvent = 'ready' | 'token' | 'error' | 'close' | 'done';
+
+export interface AmazonPayInstance extends Emitter<AmazonPayEvent> {
+  /**
+   * Invokes the Amazon Payment Modal
+   */
+  start: (amazonPayOptions: AmazonPayOptions) => void;
+
+  /**
+   * Attaches an Element to the DOM, as a child of the specified parent target.
+   *
+   */
+  attach: (defaultEventName: string, gatewayCode?: string) => AmazonPayInstance;
+
+  /**
+   * Renders Amazon Pay button to the page
+   */
+  renderButton: (element: string | HTMLElement, amazonPayRenderButtonOptions: AmazonPayOptions & AmazonPayRenderButtonOptions) => AmazonPayInstance;
```

Comment:
I'm not seeing all of these additional parameters on the `renderButton` function. It looks to me like it only accepts a `element` parameter that is a `string`

Reply:
The `renderButton` does not return anything.